mapic/mistapiconnector_app.go

```diff
@@ -610,63 +630,65 @@ func (mc *mac) triggerPushEnd(w http.ResponseWriter, r *http.Request, lines []st
 	return true
 }
 
-func (mc *mac) handleDefaultStreamTrigger(w http.ResponseWriter, r *http.Request) {
-	if r.Method != "POST" {
-		w.WriteHeader(http.StatusMethodNotAllowed)
-		w.Write([]byte("false"))
-		return
-	}
-	b, err := io.ReadAll(r.Body)
-	if err != nil {
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("false"))
-		return
-	}
-	bs := string(b)
-	lines := strings.Split(bs, "\n")
-	trigger := r.Header.Get("X-Trigger")
-	if trigger == "" {
-		glog.Errorf("Trigger not defined in request %s", bs)
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("false"))
-		return
-	}
-	mistVersion := r.Header.Get("X-Version")
-	if mistVersion == "" {
-		mistVersion = r.UserAgent()
-	}
-	glog.V(model.VERBOSE).Infof("Got request (%s) mist=%s (%d lines): `%s`", trigger, mistVersion, len(lines), strings.Join(lines, `\n`))
-	// glog.V(model.VERBOSE).Infof("User agent: %s", r.UserAgent())
-	// glog.V(model.VERBOSE).Infof("Mist version: %s", r.Header.Get("X-Version"))
-	started := time.Now()
-	doLogRequestEnd := false
-	defer func(s time.Time, t string) {
-		if doLogRequestEnd {
-			took := time.Since(s)
-			glog.V(model.VERBOSE).Infof("Request %s ended in %s", t, took)
-			metrics.TriggerDuration(t, took)
-		}
-	}(started, trigger)
-
-	switch trigger {
-	case "DEFAULT_STREAM":
-		doLogRequestEnd = mc.triggerDefaultStream(w, r, lines, trigger)
-	case "LIVE_BANDWIDTH":
-		doLogRequestEnd = mc.triggerLiveBandwidth(w, r)
-	case "CONN_CLOSE":
-		doLogRequestEnd = mc.triggerConnClose(w, r, lines, bs)
-	case "PUSH_REWRITE":
-		doLogRequestEnd = mc.triggerPushRewrite(w, r, lines, bs)
-	case "LIVE_TRACK_LIST":
-		doLogRequestEnd = mc.triggerLiveTrackList(w, r, lines, bs)
-	case "PUSH_OUT_START":
-		doLogRequestEnd = mc.triggerPushOutStart(w, r, lines, bs)
-	case "PUSH_END":
-		doLogRequestEnd = mc.triggerPushEnd(w, r, lines, bs)
-	default:
-		glog.Errorf("Got unsupported trigger: '%s'", trigger)
-		w.WriteHeader(http.StatusBadRequest)
-		w.Write([]byte("false"))
+func (mc *mac) HandleDefaultStreamTrigger() httprouter.Handle {
```

Comment:
Any chance to unit test this function (or extract some parts) and unit test?

Reply:
I'd prefer to refactor this and delete all of its trigger-handling code in favor of the `handlers/misttriggers` struct-dispatching thing; don't think it'd be worth unit testing as-is.